doc/rfcs/2023-08-17--ManagingVersions.md

```diff
@@ -0,0 +1,199 @@
+       sync hard when release is ready
+   - Con: Avoiding prior con would add non-trivial maintenance burden of
+       periodically merging master with 'next-major', and likely
+       carrying a large set of merge conflict resolutions along the way.
+   - Con: Somewhat hidden, and unlikely to see any testing
+
+3. The `cfg` option (I can't believe I'm suggesting it either)
+   - Pro: Code is 'right there' such that wide-area interface,
+       renaming, etc style changes will update next-release code as well
+       [assumes CI builds both current and next-release, which is easy]
+   - Pro: Release transition is very clear, simply remove all code where
+       `cfg` is no longer relevant.
+   - Con: It's `cfg`. It means we're carrying a bunch of code which in
+       the common case is untested and unused.
+   - Pro: Can result in space-savings for up-to-date downstream users.
+       If we are carrying code for backwards compatibility that
+       downstream folks know they won't use, they can remove that part
+       of the kernel.
+       - Con: The exponential number of kernel versions we just created.
+       - Pro: Could act as indicator for when a release is merited, i.e.
+           if we released a new version we'd shave off XX% of kernel
+           size for large enough X.
+   - Nuance: It may be worth distinguishing between "next minor" and
+       "next major". In particular, "next minor" `cfg`'d stuff can be
+       default-on, and largely no change from current practice, except
+       for this sentinel sprinkled in the codebase identifying what's
+       not in the released version [though, this could arguably also be
+       accomplished with some type of comment keyword].
+
+4. ...?
+
+[driverNum1v6]: https://github.com/tock/tock/blob/e8d0a28d86897c91b6747be357abfcfa7e86688f/capsules/src/driver.rs
+[driverNum2v1v1]: https://github.com/tock/tock/blob/44f39d7c8cf5db0038606f08640bfae670127eef/capsules/src/driver.rs
+
+
+Being more explicit about ABI changes and surface area
+------------------------------------------------------
+
+It's great that we have things like the compatibility header in the TBF,
+but if I wanted to understand which version of Tock my application
+requires, how would I figure that out? We have the list of [syscalls
+stabilized with 2.0](https://github.com/tock/tock/tree/master/doc/syscalls),
+but does that mean an app can't rely on _anything_ else?
+
+
+**Proposal:** We need something between "stable" and "unstable". Maybe
+not the [7 tiers of Python stability](https://pypi.org/classifiers/),
+but something that helps folks get a sense of 'this has been around for
+a while' versus 'this is brand new and we are still figuring it out'.
+
+
+**Proposal:** We should not stabilize whole driver interfaces, but
+rather individual syscalls within them. Low-level-debug Command 2,
+"print a number", seems pretty safe to call immutable; I don't think
+we want to lock the LLD interface yet, however.
```

Comment:
This seems like the opposite of this section, as it would make it _harder_ to know if your app relies on stable functionality or not.